platform/posix/transport/utest/reconnect_utest.c

```diff
@@ -0,0 +1,128 @@
+#include <string.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include "unity.h"
+
+/* Include paths for public enums, structures, and macros. */
+#include "transport_reconnect.h"
+
+#include "mock_unistd_api.h"
+#include "mock_stdio_api.h"
+
+static TransportReconnectParams_t reconnectParams;
+
+/* ============================   UNITY FIXTURES ============================ */
+
+/* Called before each test method. */
+void setUp()
+{
+}
+
+/* Called after each test method. */
+void tearDown()
+{
+}
+
+/* Called at the beginning of the whole suite. */
+void suiteSetUp()
+{
+}
+
+/* Called at the end of the whole suite. */
+int suiteTearDown( int numFailures )
+{
+    return numFailures;
+}
+
+/* ========================================================================== */
+
+/**
+ * @brief A helper method to make assertions on the #TransportReconnectParams_t
+ * object upon calling #Transport_ReconnectParamsReset.
+ */
+static void verifyReconnectParamsAfterReset( void )
+{
+    TEST_ASSERT_EQUAL( 0, reconnectParams.attemptsDone );
+    TEST_ASSERT_GREATER_OR_EQUAL_INT32( INITIAL_RECONNECT_BACKOFF_SECONDS,
+                                        reconnectParams.nextJitterMax );
+    TEST_ASSERT_LESS_THAN_INT32( ( INITIAL_RECONNECT_BACKOFF_SECONDS +
+                                   MAX_JITTER_VALUE_SECONDS ),
+                                 reconnectParams.nextJitterMax );
+}
+
+/**
+ * @brief Test that #Transport_ReconnectParamsReset initializes the #nextJitterMax
+ * to be equal to the some configurable backoff plus jitter.
+ */
+void test_Transport_ReconnectParamsReset_Sets_Jitter_Correctly( void )
+{
+    Transport_ReconnectParamsReset( &reconnectParams );
+    verifyReconnectParamsAfterReset();
+}
+
+/**
+ * @brief Test that #Transport_ReconnectBackoffAndSleep is able to double the
+ * max value of the next jitter and return false when all attempts are exhausted.
+ *
+ * @note #Transport_ReconnectParamsReset is expected to be called before
+ * #Transport_ReconnectBackoffAndSleep.
+ */
+void test_Transport_ReconnectBackoffAndSleep_Succeeds( void )
+{
+    bool retriesArePending;
+    int32_t expectedNextJitterMax = INITIAL_RECONNECT_BACKOFF_SECONDS;
+
+    Transport_ReconnectParamsReset( &reconnectParams );
+    verifyReconnectParamsAfterReset();
+
+    expectedNextJitterMax = reconnectParams.nextJitterMax;
+
+    while( reconnectParams.attemptsDone < MAX_RECONNECT_ATTEMPTS )
```

Comment:
This test code seems to have copied the logic inside the function under test. Instead of using the while loop, there should instead be multiple calls to `Transport_ReconnectBackoffAndSleep` (with a small values for max attempts and max backoff value so that this test can be completed in 3 or so calls)

Reply:
There is no way to change max attempts or backoff value without changing the definition in the header. Also, don't you think  these tests should run independent of the defined value?

Reply:
Instead of going through all the iterations of reaching the max threshold value, the `reconnectParams` value can be updated to reach the max attempts (after 2 sequential calls to `Transport_ReconnectBackoffAndSleep` ) so that the "max attempts" reached case is simulated.

I prefer that the test code doesn't copy the implementation code!

Reply:
It is simply expecting each iteration of the retry... You simply do 2 sequential calls, and any test that simply increases attempts done will still pass.